jaeger-core/src/main/java/com/uber/jaeger/senders/UDPSender.java

```diff
@@ -85,14 +92,20 @@ int getSizeOfSerializedSpan(Span span) throws SenderException {
    */
   @Override
   public int append(Span span) throws SenderException {
-    int spanSize = getSizeOfSerializedSpan(span);
+    if (process == null) {
```

Comment:
I just realized, there's another issue here. The previous Zipkin based code was able to compute the cumulative size of the message incrementally by sizing each span and adding to the overhead constant. But in jaeger.thrift model we have another variable, the process tags. Good news is we do not allow process tags to be changed post-construction, so we can pre-calculate their thrift size. But the overall calculation is getting more complex because Compact encoding uses varint encoding for things like the size of collections. So we need to be careful with the max-overhead constant. Ideally there should be a unit test to ensure that the final serialized buffer size does not exceed 65000, or it may be dropped by the agent.